diff --git a/node_modules/@strapi/plugin-users-permissions/dist/server/services/providers-registry.js b/node_modules/@strapi/plugin-users-permissions/dist/server/services/providers-registry.js
index 9063695..595f47a 100644
--- a/node_modules/@strapi/plugin-users-permissions/dist/server/services/providers-registry.js
+++ b/node_modules/@strapi/plugin-users-permissions/dist/server/services/providers-registry.js
@@ -509,7 +509,30 @@ function requireProvidersRegistry() {
                         };
                     });
                 }
-            }
+            },
+            unified: {
+                enabled: false,
+                icon: '',
+                grantConfig: {
+                    key: '',
+                    secret: '',
+                    callback: `${baseURL}/unified/callback`,
+                    scope: [
+                        'openid'
+                    ],
+                },
+                async authCallback({ accessToken }) {
+                    const unified = purest({ 
+                        provider: 'unified' 
+                    });
+                    return unified.get('userinfo').auth(accessToken).request().then(({ body }) => {
+                        return {
+                            username: `${body.firstName} ${body.lastName}`,
+                            email: body.loginName,
+                        };
+                    });
+                },
+            },
         });
     providersRegistry = ()=>{
         const purest = require$$4;
